carta/image.py

```diff
@@ -42,39 +42,32 @@ def __init__(self, session, image_id, file_name):
         self._frame = Macro("", self._base_path)
 
     @classmethod
-    def new(cls, session, path, hdu, append, complex, expression):
+    def new(cls, session, directory, file_name, hdu, append, image_arithmetic):
```

Comment:
When I wrote this I thought that I couldn't use the validation decorator here, but I think that it is actually possible. Please try adding the validation decorator (put it between the `@classmethod` decorator and the function definition). This includes replacing the handwritten types with `{0}` and so on.

Reply:
Done in c0838d8.
Nevertheless, I just added validators for `directory`, `file_name`, `hdu`, `append` and `image_arithmetic`. Do I need to add validator for `session`?

Reply:
Yes, the validation decorator parameters have to match the function parameters, or everything will be shifted up by one and nothing will work. You should be able to use `InstanceOf` and the session class.

I think I initially decided not to use validation here because the function isn't meant to be used directly, but it's exposed and it could be, so it's probably a good idea. I haven't previously tested chaining the decorators like this, so please be sure to test calling the function directly to make sure that everything is working.